### ISSUE TITLE
Why x - x.detach() Instead of Referencing ref_per_token_logps?

### ISSUE THREAD

**Issue by buchuoer** [tulerfeng/Video-R1#62]
I have a question regarding a detail in the code: In the file src/r1-v/src/open_r1/trainer/vllm_grpo_trainer_modified.py, at line 830:
```
#     x - x.detach() allows for preserving gradients from x  
per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1)  
per_token_loss = -(per_token_loss - self.beta * per_token_kl)  
# per_token_loss = -per_token_loss  
loss = ((per_token_loss * completion_mask).sum(dim=1) / completion_mask.sum(dim=1)).mean()  
```
Why is x - x.detach() used directly here? According to the original paper, shouldn't it be per_token_logps - ref_per_token_logps.detach() instead?
